src/plugins/discord/fetcher.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import * as Model from "./models";
+
+type FetchEndpoint = (endpoint: string) => Promise<any>;
+
+/**
+ * Provide the Guild ID to fetch against, and the 'limit'
+ * parameter when fetching GuildMembers, Messages, and Reactions.
+ *
+ */
+type FetchOptions = {|
+  guildId: Model.Snowflake,
```

Comment:
My inclination was that `guildId` is a constant, and there's no reason to parameterize a constant; just increases the surface area on which we could error. 

That caveat aside, I updated according to your suggestion.

Reply:
Thanks for the change!

I found the previous thread now: https://github.com/sourcecred/sourcecred/pull/1703#discussion_r392710220

> My rationale for this `options` object:
> 
> ```js
> type FetchOptions = {|
>   guildId: number,
>   membersLimit: number,
>   messagesLimit: number,
>   reactionsLimit: number,
> |};
> ```
> 
> These will be constants for the duration of a `Fetchers` use-case. I don't think we should parameterize variables that won't change, and it makes it easier to reason about the function parameters that _will_ change by keeping constants out of function signatures. E.g., we're not going to use the same `Fetcher` to fetch different `guildId`s. It also reduces the surface area where mistakes could occur by providing these constants in one place (at the classes' initialization) instead of several (whenever we need to call a method that requires `guildId`).
> 
> Not a strong opinion however. If we wanted to remove the `guildId` from the `options` object, I'd think we want to do the same with the `limit` variables as well.
> 
> What do you think?

Gotcha! Yeah I think there's no technical reason why Fetcher should require that `guildId` doesn't change. By doing that we're making Fetcher responsible for the use-case of Fetcher. That responsibility should be with what's using Fetcher.

If the use-case is mirroring a guild's data. Then the guild mirroring function should add constraints like `guildId` is constant for the whole of this process.

An example of a different use-case is to answer the question "with this bot token, which guilds do I have permissions for?". We might use such a query for validating the token, or to help users configure the plugin. It wouldn't make sense if this required a constant `guildId` be given to the constructor.